### PR TITLE
chore(deps): bump ipfs networking deps

### PR DESCRIPTION
**Pull request by walldiss**
## Overview
Main change is update of github.com/ipfs/go-libipfs , that is now located in github.com/ipfs/boxo 

Updates libs and their dependancies:
- github.com/libp2p/go-libp2p-kad-dht
- github.com/libp2p/go-libp2p-routing-helpers
 - github.com/ipfs/go-block-format 
- 	github.com/ipfs/go-blockservice 
 - 	github.com/ipfs/go-ipld-format
 - 	github.com/ipfs/go-merkledag 
 - 	github.com/ipld/go-car 
 - 	github.com/multiformats/go-multihash 

Supersedes:
https://github.com/celestiaorg/celestia-node/pull/2268
https://github.com/celestiaorg/celestia-node/pull/2378
https://github.com/celestiaorg/celestia-node/pull/2379
https://github.com/celestiaorg/celestia-node/pull/1957

### CODE REVIEW

**Comment by Wondertan** [comment]
snyk is complaining

**Comment by Wondertan** [comment]
Per sync discussion:
* Needs one unit test to fix. Likely because dht is smarter now 
* This updates to deprecated versions, and we will be able to fully migrate to boxo once https://github.com/ipfs/boxo/pull/407 and (follow-ups) described there are merged.

File: share/ipld/nmt.go [hunk 1]
```diff
@@ -74,6 +74,7 @@ func GetNode(ctx context.Context, bGetter blockservice.BlockGetter, root cid.Cid
 	if err != nil {
 		var errNotFound ipld.ErrNotFound
 		if errors.As(err, &errNotFound) {
+			fmt.Println("BLOCK NOT FOUND")
```
Comment (renaynay) [inline share/ipld/nmt.go:77]: print

**Comment by Wondertan** [comment]
New release https://github.com/libp2p/go-libp2p-kad-dht/releases/tag/v0.24.3

**Comment by Wondertan** [comment]
Taking ower this PR

**Comment by renaynay** [comment]
Works now after rebasing on #2560

**Comment by Wondertan** [comment]
Soon, https://github.com/ipfs/boxo/pull/407 will be released so we can fully migrate to boxo

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — APPROVED** [review]
already approved and recent changes lgtm

**Pull request merged**